### PR TITLE
[webview_flutter] Add a getTitle method to WebViewController

### PR DESCRIPTION
**Pull request by amirh**
## Description

Allows inspecting the title of the currently loaded page.

## Related Issues

https://github.com/flutter/flutter/issues/37177

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by ened** [comment]
Could this possibly be a `Stream<String>` as the title may whenever the web view loads the next page?

**Comment by ened** [comment]
> Could this possibly be a `Stream<String>` as the title may change whenever the web view loads the next page?

Actually no need, there's a page listener that's called when a (or any further) page is loaded, so developers could use that. :-) Ignore my original comment.

**Comment by amirh** [comment]
> Could this possibly be a `Stream<String>` as the title may change whenever the web view loads the next page?

Good point, thanks!

I think a broadcast stream makes sense for this case, though I think we would still need this getter for the "current" value (as broadcast streams drop values if there are no listeners).

WDYT?

**Comment by amirh** [comment]
> > Could this possibly be a `Stream<String>` as the title may change whenever the web view loads the next page?
> 
> Actually no need, there's a page listener that's called when a (or any further) page is loaded, so developers could use that. :-) Ignore my original comment.

Your comment is still valid as the title might change without a new page being loaded (e.g javascript code can change the title).

I do think it still makes sense to to have this getter as I commented above though.

**Comment by ened** [comment]
If JS code can change the title, we'd need to fiddle with the html to get it, unless `WebView` has some sort of listener we can hook into. 

From my point of view, the current PR (to get the title) provides enough of a building block to build a stream of titles later when combining this with the page listener and perhaps other methods (even a `Timer.periodic`..).

Thx @amirh.

**Comment by amirh** [comment]
> If JS code can change the title, we'd need to fiddle with the html to get it, unless `WebView` has some sort of listener we can hook into.
> 
> From my point of view, the current PR (to get the title) provides enough of a building block to build a stream of titles later when combining this with the page listener and perhaps other methods (even a `Timer.periodic`..).
> 
> Thx @amirh.

Yes such listeners are available on Android and iOS.
Anyway going to land this PR as-is it seems that we all agree this getter makes sense.

File: packages/webview_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
+## 0.3.12
+
+* Added a getTitle getter to WebViewController.
+
 ## 0.3.11+6
 
 * Calling destroy on Android webview when flutter webview is getting disposed.
+>>>>>>> master
```
Comment (ened) [inline packages/webview_flutter/CHANGELOG.md]: @amirh saw this?

Comment (amirh) [reply packages/webview_flutter/CHANGELOG.md]: oops

**Pull request merged**